### ISSUE TITLE
Minimum python version

### ISSUE THREAD

**Issue by delneg** [fable-compiler/Fable.Python#12]
What version will this be working starting from ?
I.e. in https://github.com/python/typeshed/blob/master/stdlib/uuid.pyi
there are separate type for version 3.7 and above and others. So, it's best to determine which this lib will support right away.

**Comment by dbrattli** [comment]
@hlizard Coconut looks like a really nice project, but I don't think it's the right abstraction for Fable. Python 2 is out of the question anyways, and solving the Python 3 problem would only be replaced with a similar Coconut version problem.

**Comment by smoothdeveloper** [comment]
Just to add to the discussion, right now Fable python target compiles code that doesn't seem to load in python 3.9.7, providing this error message:

> TypeError: 'ABCMeta' object is not subscriptable

Stemming from

```python
from .types import FSharpRef, Record
```

As far as I could search, it is from an error related to type annotation in conjunction with the @dataclass attributes that are put in the fable.core standard library for python, and it seems to trip the 3.9.7 parser.

I don't know exactly in which release it starts to work, it does work on 3.11.2, and I vaguely remember seeing "fixed in 3.10" but I can't find the PEP nor the stackoverflow answer that pointed me there.

For this reason, I'd suggest not mentioning python 3.9 or bumping this to a higher version in the readme of this repository.

As for the bindings in this repository and which version of the python library to support, that seems to be a big can of worm, python seems to be evolving fast, some of my own code written few months back starts giving me deprecation warnings after doing a minor bump of the python version.

**Comment by dbrattli** [comment]
Yes, agree. Removing support for Python 3.9 in Fable. Will update the readme

**Comment by dbrattli** [comment]
Fable.Python will require Python 3.12+ when Fable v5 is released
